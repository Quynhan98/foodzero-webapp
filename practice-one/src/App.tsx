import { Header } from '@components/header';
import { ManagementSection } from '@components/management-section';
import { ProductSection } from '@components/product-section';
import { WorkSection } from '@components/work-section';
import '../node_modules/bootstrap/dist/css/bootstrap.min.css';
import './styles/main.css';

function App() {
  return (
    <div className="App container">
      <Header />
      <WorkSection />
      <ProductSection />
      <ManagementSection />
    </div>
  );
}

export default App;
