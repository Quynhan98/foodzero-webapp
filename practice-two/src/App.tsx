import { Error } from '@components/common/error';
import { Popup } from '@components/common/popup';
import { Form } from '@components/form';
import { Header } from '@components/header';
import { Books } from '@components/list-book';
import { SERVER_MESSAGES } from '@constant/messages';
import URL_PAGE from '@constant/url';
import { formValidate, ValidationResult } from '@helpers/validation-form';
import useGetData from '@hooks/useGetData';
import { Book } from '@interface/book';
import { postData } from '@services/fetch-api';
import { useState } from 'react';
import './styles/main.css';

const App = (): JSX.Element => {
  const [isOpen, setIsOpen] = useState(false);
  const { data, error, setData, setError } = useGetData<Book>(URL_PAGE);

  const handelToggleForm = (): void => {
    setIsOpen(!isOpen);
  };

  const handleCreate = async (book: Book): Promise<void> => {
    try {
      const validateFormCreate: ValidationResult = formValidate(book);

      if (validateFormCreate.isValid) {
        await postData(book, URL_PAGE);

        setData([...data, book]);
        handelToggleForm();
      }
    } catch (error) {
      setError(SERVER_MESSAGES.SERVER_RESPONSE_ERROR);
      handelToggleForm();
    }
  };

  const closeError = (): void => {
    setError('');
  };

  return (
    <div className="container">
      <Header toggleForm={handelToggleForm} />
      <Books books={data} />
      {isOpen && (
        <Popup handleClose={handelToggleForm}>
          <Form onCreate={handleCreate} onHandleClose={handelToggleForm} />
        </Popup>
      )}
      {error && <Error onHandleClick={closeError}>{error}</Error>}
    </div>
  );
};

export default App;
