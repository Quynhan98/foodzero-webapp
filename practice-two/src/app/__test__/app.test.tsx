import { fireEvent, render, screen } from '@testing-library/react';
import App from '@app/App';

describe('Test App', () => {
  test('App matchers DOM Snapshot', () => {
    const component = render(<App />);

    expect(component).toMatchSnapshot();
  });

  test('App should call Form Create', () => {
    render(<App />);

    fireEvent.click(screen.getByText('Add New Book'));
    expect(screen.getByText('Create New Book')).toHaveTextContent('Create New Book');
    fireEvent.click(screen.getByText('Cancel'));
    expect(screen.getByTestId('app')).not.toHaveTextContent('Create New Book');
  });
});
