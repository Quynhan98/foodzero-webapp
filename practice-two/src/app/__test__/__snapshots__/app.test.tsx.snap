// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App App matchers DOM Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
        data-testid="app"
      >
        <header
          class="page-header"
        >
          <h2
            class="heading"
          >
            A list of books
          </h2>
          <div
            class="nav-btn"
          >
            <button
              class="button btn-primary btn-large"
              type="button"
            >
              Add New Book
            </button>
          </div>
        </header>
        <div
          class="filter-search"
        >
          <input
            class="search-input"
            data-testid="search"
            placeholder="Search..."
            type="search"
          />
        </div>
        <div
          class="book-list"
        >
          <div
            class="card-group"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      data-testid="app"
    >
      <header
        class="page-header"
      >
        <h2
          class="heading"
        >
          A list of books
        </h2>
        <div
          class="nav-btn"
        >
          <button
            class="button btn-primary btn-large"
            type="button"
          >
            Add New Book
          </button>
        </div>
      </header>
      <div
        class="filter-search"
      >
        <input
          class="search-input"
          data-testid="search"
          placeholder="Search..."
          type="search"
        />
      </div>
      <div
        class="book-list"
      >
        <div
          class="card-group"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
